File: cross-platform/utils/WebHookUtils.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class WebHookUtils {
  String token = "";
  Future<String> getTestEmail() async {
    if (token == "") {
      final tokenResponse =
          await http.post(Uri.parse('https://webhook.site/token'));
      final tokenData = jsonDecode(tokenResponse.body);
      token = tokenData['uuid'];
    }
    return "$token@email.webhook.site";
  }

  Future<String> getSignUpToken() async {
    final tokenResponse = await http
        .get(Uri.parse("https://webhook.site/token/$token/request/latest"));
    final tokenData = jsonDecode(tokenResponse.body);
    return tokenData['headers']['subject'][0].substring(0, 8);
  }

  Future<String> getTokenString() async {
    if (token == "") {
      final tokenResponse =
          await http.post(Uri.parse('https://webhook.site/token'));
      final tokenData = jsonDecode(tokenResponse.body);
      token = tokenData['uuid'];
    }
    return token;
  }
}

File: web-frontend/cypress/WebHookUtils.js
// Define the WebHookUtils class
export default class WebHookUtils {
    // Initialize the token property
    constructor() {
        this.token = '';
    }

    // Define the getTestEmail method
    async getTestEmail() {
        // If the token is empty, request a new one
        if (this.token == '') {
            const tokenResponse = await fetch('https://webhook.site/token', {
                method: 'POST',
            });
            const tokenData = await tokenResponse.json();
            this.token = tokenData['uuid'];
        }
        // Return the test email address
        return `${this.token}@email.webhook.site`;
    }

    // Define the getSignUpToken method
    async getSignUpToken() {
        // Get the latest request from the token
        const tokenResponse = await fetch(
            `https://webhook.site/token/${this.token}/request/latest`
        );
        const tokenData = await tokenResponse.json();
        // Return the sign up token from the subject header
        return tokenData['headers']['subject'][0].substring(0, 8);
    }

    // Define the getTokenString method
    async getTokenString() {
        // If the token is empty, request a new one
        if (this.token == '') {
            const tokenResponse = await fetch('https://webhook.site/token', {
                method: 'POST',
            });
            const tokenData = await tokenResponse.json();
            this.token = tokenData['uuid'];
        }
        // Return the token string
        return this.token;
    }
}
